endpoints-framework/src/main/java/com/google/api/server/spi/config/model/MapSchemaFlag.java

```diff
@@ -0,0 +1,64 @@
+package com.google.api.server.spi.config.model;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * These flags control the behavior of the schema generators regarding Map types.
+ *
+ * By default, schema generation uses "additionalProperties" in JsonSchema to describe Map types
+ * (both for Discovery and OpenAPI), with a proper description of the value types.
+ * This mode supports key types that can be serialized from / to String, and supports any value
+ * type except array-like ones (see {@link MapSchemaFlag#SUPPORT_ARRAYS_VALUES} for more details).
+ * In previous versions of Cloud Endpoints, Maps were always represented using the untyped "JsonMap"
+ * object (see {@link com.google.api.server.spi.config.model.SchemaRepository#MAP_SCHEMA}).
+ *
+ * To enable one of these enum flags, you can either:
+ * - Set system property "endpoints.mapSchema." + systemPropertyName to any value except a falsy one
```

Comment:
systemPropertyName() ?
MapSchemaFlag.systemPropertyName() ?
(It wasn't clear for me at first read what was this systemPropertyName since I read the description before the code)

Reply:
The is indeed a confusion here, as we have constructor param and field with same name but different values. Renamed constructor param to systemPropertySuffix, and improved documentation.